modules/3d/test/test_octree.cpp

```diff
@@ -0,0 +1,112 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+using namespace cv;
+
+class OctreeTest: public::testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        pointCloudSize = 1000;
+        maxDepth = 4;
+
+        int scale;
+        Point3i pmin, pmax;
+        scale = 1<<20;
+        pmin = Point3i(-scale, -scale, -scale);
+        pmax = Point3i(scale, scale, scale);
+
+        RNG& rng_Point = theRNG(); // set random seed for fixing output 3D point.
+
+        // Generate 3D PointCloud
+        for(int i = 0; i < pointCloudSize; i++)
+        {
+            float _x = (float)rng_Point.uniform(pmin.x, pmax.x)/scale;
+            float _y = (float)rng_Point.uniform(pmin.y, pmax.y)/scale;
+            float _z = (float)rng_Point.uniform(pmin.z, pmax.z)/scale;
```

Comment:
code generates values from the range `[-1; 1]`.

Perhaps this range should be extended to `[-10; 10]`.